lib/vm.js

```diff
@@ -84,14 +84,36 @@ function validateString(prop, propName) {
     throw new ERR_INVALID_ARG_TYPE(propName, 'string', prop);
 }
 
+function validateBool(prop, propName) {
+  if (prop !== undefined && typeof prop !== 'boolean')
+    throw new ERR_INVALID_ARG_TYPE(propName, 'boolean', prop);
+}
+
+function validateObject(prop, propName) {
```

Comment:
Should we rather name this function `validateObjectOrUndefined`?

Reply:
none of the others are named like that, and it seems fairly self-explanatory